src/lib/util/zippath.cpp

```diff
@@ -223,21 +223,21 @@ std::string zippath_parent_basename(const char *path)
  * @return  A std::string&amp;
  */
 
-std::string &zippath_combine(std::string &dst, const char *path1, const char *path2)
+std::string &zippath_combine(std::string &dst, const std::string &path1, const std::string &path2)
 {
-	if (!strcmp(path2, "."))
+	if (path2 == ".")
 	{
 		dst.assign(path1);
 	}
-	else if (!strcmp(path2, ".."))
+	else if (path2 == "..")
 	{
 		dst = zippath_parent(path1);
 	}
 	else if (osd_is_absolute_path(path2))
 	{
 		dst.assign(path2);
 	}
-	else if ((path1[0] != '\0') && !is_path_separator(path1[strlen(path1) - 1]))
+	else if ((path1[0] != '\0') && !is_path_separator(path1[path1.length() - 1]))
```

Comment:
You should check `str.empty()` rather than looking for NUL in first element, and use `str.back()` or `*str.rbegin()` rather than checking element `length() - 1`.